timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -481,12 +481,13 @@ void __global__ k_compute_nonbonded_target_atom_energies(
         RealType ci_z = coords[atom_i_idx * 3 + 2];
 
         int atom_j_idx = blockIdx.x * blockDim.x + threadIdx.x;
-        // Zero out the energy buffer
-        block_energy_buffer[threadIdx.x] = 0;
-        while (atom_j_idx < N) {
+        // All threads in the threadblock must loop to allow for __syncthreads() and the row accumulation.
+        while (atom_j_idx - threadIdx.x < N) {
```

Comment:
nit: slightly easier to comprehend for me as
```suggestion
        while (blockIdx.x * blockDim.x < N) {
```

Reply:
This would result in an infinite loop since `blockIdx.x * blockDim.x` is constant for each kernel

Reply:
Ah, right, sorry. I missed that `atom_j_idx` is being modified in the loop.